### PR TITLE
Native Branch: Replace ACE/get_opt with custom ArgparseForServer

### PR DESCRIPTION
**Pull request by 0blu**
## 🍰 Pullrequest
Replace ACE/get_opt with custom ArgparseForServer

### CODE REVIEW

**Comment by 0blu** [comment]
I take some effort to make a nice git history, please dont squash merge them :(

**Comment by ratkosrb** [comment]
But we always squash merge in vmangos. Otherwise you can't revert cleanly, or go back to revision before I clicked merge, because the commits will be spread out. We never do regular merge in vmangos. When this branch is ready the whole native branch will be squash merged into development with commit named "Remove usage of ACE framework".

**Comment by 0blu** [comment]
Ah okay, I see, reversion is actually useful.

File: src/mangosd/Main.cpp [hunk 1]
```diff
@@ -59,127 +59,58 @@ DatabaseType LogsDatabase;                                  // Accessor to the l
...
-    case 's':
+    case ServiceDaemonAction::Stop:
         stopDaemon();
         break;
-    }
 #endif
+    }
 
-    sLog.Out(LOG_BASIC, LOG_LVL_MINIMAL, "Core revision: %s [world-daemon]", _FULLVERSION);
+    sLog.Out(LOG_BASIC, LOG_LVL_MINIMAL, "Core revision: %s [realm-daemon]", _FULLVERSION);
```
Comment (0blu) [inline src/mangosd/Main.cpp:113]: Woops

**Pull request merged**